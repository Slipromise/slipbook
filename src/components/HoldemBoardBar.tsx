"use client";

import { ComponentProps } from "react";
import PokerCard from "./PokerCard";
import styles from "@/styles/components/HoldemBoardBar.module.scss";
import { useSpring, animated } from "@react-spring/web";

type PokerCard = ComponentProps<typeof PokerCard>["value"];

type Props = {
  cards?: PokerCard[];
  pot: number;
  subtitle: string;
  locale?: string;
};

function HoldemBoardBar({ cards, pot, subtitle, locale = "en-US" }: Props) {
  const springValues = useSpring({
    pot,
    config: {
      duration: 300,
    },
  });
  return (
    <div className={styles.container}>
      <div className={styles.header}>
        {cards?.map((item) => (
          <PokerCard value={item} key={item} />
        ))}
      </div>
      <div className={styles.body}>
        <span>POT</span>{" "}
        <animated.span>
          {springValues.pot.to((n) =>
            new Intl.NumberFormat(locale, {
              style: "currency",
              // TODO: check formatting
              currency: "USD",
            }).format(n)
          )}
        </animated.span>
      </div>
      <div className={styles.footer}>
        <span>{subtitle}</span>
      </div>
    </div>
  );
}

export default HoldemBoardBar;
