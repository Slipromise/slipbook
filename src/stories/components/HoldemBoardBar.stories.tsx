import HoldemBoardBar from "@/components/HoldemBoardBar";
import type { Meta, StoryObj } from "@storybook/react";

const meta: Meta<typeof HoldemBoardBar> = {
  component: HoldemBoardBar,
  tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof HoldemBoardBar>;

export const PreFlop: Story = {
  args: {
    pot: 100000,
    subtitle: "2024 XXXX EVENTS - final table",
  },
};

export const Flop: Story = {
  args: {
    cards: ["C2", "H2", "DJ"],
    pot: 300000,
    subtitle: "2024 XXXX EVENTS - final table",
  },
};

export const Turn: Story = {
  args: {
    cards: ["C2", "H2", "DJ", "SK"],
    pot: 1000000,
    subtitle: "2024 XXXX EVENTS - final table",
  },
};

export const River: Story = {
  args: {
    cards: ["C2", "H2", "DJ", "SK", "SQ"],
    pot: 10000000,
    subtitle: "2024 XXXX EVENTS - final table",
  },
};
