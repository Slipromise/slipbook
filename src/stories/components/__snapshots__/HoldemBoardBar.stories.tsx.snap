// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stories/components/HoldemBoardBar Flop smoke-test 1`] = `
<div class="HoldemBoardBar_container__5s_so">
  <div class="HoldemBoardBar_header__i43Uj">
    <div class="PokerCard_container__qevio"
         data-card="C2"
    >
      <span data-front-card="true">
        2
      </span>
      <span data-back-card="true">
      </span>
    </div>
    <div class="PokerCard_container__qevio"
         data-card="H2"
    >
      <span data-front-card="true">
        2
      </span>
      <span data-back-card="true">
      </span>
    </div>
    <div class="PokerCard_container__qevio"
         data-card="DJ"
    >
      <span data-front-card="true">
        J
      </span>
      <span data-back-card="true">
      </span>
    </div>
  </div>
  <div class="HoldemBoardBar_body__fI1w0">
    <span>
      POT
    </span>
    <span>
      300,000
    </span>
  </div>
  <div class="HoldemBoardBar_footer__pOKcU">
    <span>
      2024 XXXX EVENTS - final table
    </span>
  </div>
</div>
`;

exports[`stories/components/HoldemBoardBar PreFlop smoke-test 1`] = `
<div class="HoldemBoardBar_container__5s_so">
  <div class="HoldemBoardBar_header__i43Uj">
  </div>
  <div class="HoldemBoardBar_body__fI1w0">
    <span>
      POT
    </span>
    <span>
      100,000
    </span>
  </div>
  <div class="HoldemBoardBar_footer__pOKcU">
    <span>
      2024 XXXX EVENTS - final table
    </span>
  </div>
</div>
`;

exports[`stories/components/HoldemBoardBar River smoke-test 1`] = `
<div class="HoldemBoardBar_container__5s_so">
  <div class="HoldemBoardBar_header__i43Uj">
    <div class="PokerCard_container__qevio"
         data-card="C2"
    >
      <span data-front-card="true">
        2
      </span>
      <span data-back-card="true">
      </span>
    </div>
    <div class="PokerCard_container__qevio"
         data-card="H2"
    >
      <span data-front-card="true">
        2
      </span>
      <span data-back-card="true">
      </span>
    </div>
    <div class="PokerCard_container__qevio"
         data-card="DJ"
    >
      <span data-front-card="true">
        J
      </span>
      <span data-back-card="true">
      </span>
    </div>
    <div class="PokerCard_container__qevio"
         data-card="SK"
    >
      <span data-front-card="true">
        K
      </span>
      <span data-back-card="true">
      </span>
    </div>
    <div class="PokerCard_container__qevio"
         data-card="SQ"
    >
      <span data-front-card="true">
        Q
      </span>
      <span data-back-card="true">
      </span>
    </div>
  </div>
  <div class="HoldemBoardBar_body__fI1w0">
    <span>
      POT
    </span>
    <span>
      10,000,000
    </span>
  </div>
  <div class="HoldemBoardBar_footer__pOKcU">
    <span>
      2024 XXXX EVENTS - final table
    </span>
  </div>
</div>
`;

exports[`stories/components/HoldemBoardBar Turn smoke-test 1`] = `
<div class="HoldemBoardBar_container__5s_so">
  <div class="HoldemBoardBar_header__i43Uj">
    <div class="PokerCard_container__qevio"
         data-card="C2"
    >
      <span data-front-card="true">
        2
      </span>
      <span data-back-card="true">
      </span>
    </div>
    <div class="PokerCard_container__qevio"
         data-card="H2"
    >
      <span data-front-card="true">
        2
      </span>
      <span data-back-card="true">
      </span>
    </div>
    <div class="PokerCard_container__qevio"
         data-card="DJ"
    >
      <span data-front-card="true">
        J
      </span>
      <span data-back-card="true">
      </span>
    </div>
    <div class="PokerCard_container__qevio"
         data-card="SK"
    >
      <span data-front-card="true">
        K
      </span>
      <span data-back-card="true">
      </span>
    </div>
  </div>
  <div class="HoldemBoardBar_body__fI1w0">
    <span>
      POT
    </span>
    <span>
      1,000,000
    </span>
  </div>
  <div class="HoldemBoardBar_footer__pOKcU">
    <span>
      2024 XXXX EVENTS - final table
    </span>
  </div>
</div>
`;
